File: src/test/java/com/github/xumeng/java/problems/easy/problem0326/Solution0326Test.java
package com.github.xumeng.java.problems.easy.problem0326;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class Solution0326Test {

    private final Solution0326 solution = new Solution0326();

    @Test
    public void test1() {
        int n = -1;
        boolean isPowerOfThree = solution.isPowerOfThree(n);
        assertFalse(isPowerOfThree);
    }

    @Test
    public void test2() {
        int n = 0;
        boolean isPowerOfThree = solution.isPowerOfThree(n);
        assertFalse(isPowerOfThree);
    }

    @Test
    public void test3() {
        int n = 9;
        boolean isPowerOfThree = solution.isPowerOfThree(n);
        assertTrue(isPowerOfThree);
    }

    @Test
    public void test4() {
        int n = 27;
        boolean isPowerOfThree = solution.isPowerOfThree(n);
        assertTrue(isPowerOfThree);
    }

    @Test
    public void test5() {
        int n = 45;
        boolean isPowerOfThree = solution.isPowerOfThree(n);
        assertFalse(isPowerOfThree);
    }

    @Test
    public void test6() {
        int n = 2147483647;
        boolean isPowerOfThree = solution.isPowerOfThree(n);
        assertFalse(isPowerOfThree);
    }

}

File: src/test/kotlin/com/github/xumeng/kotlin/problems/easy/problem0326/Solution0326Test.kt
package com.github.xumeng.kotlin.problems.easy.problem0326

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class Solution0326Test {

    private val solution = Solution0326()


    @Test
    fun test1() {
        val n = -1
        val isPowerOfThree = solution.isPowerOfThree(n)
        assertFalse(isPowerOfThree)
    }

    @Test
    fun test2() {
        val n = 0
        val isPowerOfThree = solution.isPowerOfThree(n)
        assertFalse(isPowerOfThree)
    }

    @Test
    fun test3() {
        val n = 9
        val isPowerOfThree = solution.isPowerOfThree(n)
        assertTrue(isPowerOfThree)
    }

    @Test
    fun test4() {
        val n = 27
        val isPowerOfThree = solution.isPowerOfThree(n)
        assertTrue(isPowerOfThree)
    }

    @Test
    fun test5() {
        val n = 45
        val isPowerOfThree = solution.isPowerOfThree(n)
        assertFalse(isPowerOfThree)
    }

    @Test
    fun test6() {
        val n = 2147483647
        val isPowerOfThree = solution.isPowerOfThree(n)
        assertFalse(isPowerOfThree)
    }
}